src/contracts/middleware/IndexRegistry.sol

```diff
@@ -0,0 +1,201 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+
+import "../interfaces/IIndexRegistry.sol";
+import "../interfaces/IRegistryCoordinator.sol";
+import "../libraries/BN254.sol";
+import "forge-std/Test.sol";
+
+
+contract IndexRegistry is IIndexRegistry, Test {
+
+    IRegistryCoordinator public registryCoordinator;
+
+    // list of all unique registered operators
+    bytes32[] public globalOperatorList;
+
+    // mapping of operatorId => quorumNumber => index history of that operator
+    mapping(bytes32 => mapping(uint8 => OperatorIndex[])) public operatorIdToIndexHistory;
+    // mapping of quorumNumber => history of numbers of unique registered operators
+    mapping(uint8 => OperatorIndex[]) public totalOperatorsHistory;
+
+    modifier onlyRegistryCoordinator() {
+        require(msg.sender == address(registryCoordinator), "IndexRegistry.onlyRegistryCoordinator: caller is not the registry coordinator");
+        _;
+    }
+
+    constructor(
+        IRegistryCoordinator _registryCoordinator
+    ){
+        registryCoordinator = _registryCoordinator;
+    }
+
+    /**
+     * @notice Registers the operator with the specified `operatorId` for the quorums specified by `quorumBitmap`.
+     * @param operatorId is the id of the operator that is being registered
+     * @param quorumNumbers is the quorum numbers the operator is registered for
+     * @dev access restricted to the RegistryCoordinator
+     * @dev Preconditions:
```

Comment:
I think whenever we say 'preconditions', we should be clear about if they are assumed to be checked (so the input is trusted) or if the function checks them itself / will reject inputs that don't meet the conditions.

Reply:
I'm OK with, e.g. any of these wordings:
`preconditions (these are assumed)`
or
`preconditions (function will revert if conditions not met)`
or
`assumptions on inputs`
or
`checks performed on inputs`

Reply:
good point https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/4b8f0edb7ec925effd8c53f0bfd74e7f255a5d64